spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClients.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.client;
+
+import org.candlepin.dto.api.v1.CertificateDTO;
+
+/**
+ * Static factory created for easy selection of appropriate authentication method.
+ */
+public final class ApiClients {
```

Comment:
The name ApiClients is confusing to me, and do we need this static class? Can we expect our spec tests to extend the SpecTestFixture class which already has the ApiClientFactory? If we do, we could move these methods to the abstract class and let the spec tests inherit them for free.

Reply:
The point here is to have a well defined entry point for the client separate from the test code. Treating the client as a library will help us keeping it maintainable. Abstract parent of all spec test is likely to become a dumping ground for everything related like DatabaseTestFixture.

Reply:
While I agree that DBTestFixture is somewhat bloated with stuff that could go elsewhere, I feel all the scaffolding we're building here is the same class of problem just spread out more. Like we have ApiClients (plural), ApiClient (singular), a pile of interceptors, and a bunch of other stuff needed for each individual test. In a lot of ways, we're moving the pain from bloat on a class that we just use for a certain type of test, to needing a ton of boilerplate in the tests themselves. Eventually that boilerplate is going to be gathered up in a single class again, and we'll be right back to the DBTestFixture problem.